Optimize scroll-to-bottom

AndyScherzinger opened the pull request:
* Replace imageButton with MaterialButton (to also have a ripple effect)
* Apply primary tonal dynamic coloring
* Replace icon with a version with higher emphasis (translates to thickness in M3)
* resize 44dp ->48dp as minSize for accessibility
* raise end-padding from 6dp tp 8dp to stay in the typical Material Design spacings

Signed-off-by: Andy Scherzinger <info@andy-scherzinger.de>

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
![Screenshot_20230207_103440](https://user-images.githubusercontent.com/1315170/217207310-4d4d2066-4578-4b16-bb93-de125e26d028.png) | ![Screenshot_20230208_093815](https://user-images.githubusercontent.com/1315170/217477976-561c3536-4e95-4e36-85e5-24e1b1344075.png)

### 🚧 TODO

- [x] all good to go

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [x] 🔖 Capability is checked or not needed 
- [x] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [x] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

AndyScherzinger commented:
@rapterjet2004 just for your interest, this is my follow-up PR to your scroll-down work adding the theming optimizations as mentioned in the review. In case your are interested :)

AndyScherzinger commented:
Also CC @nimishavijay for the visual optimizations check 💙

nimishavijay commented:
Looks super nice! :) Looking at the [m3 FAB specs](https://m3.material.io/components/floating-action-button/specs) do we also want to:
- Make it a rectangle with `border-radius: 12dp` instead of a circular button? 
- Add a shadow (I would this is not necessary since it is not a primary action like creating a conversation as the examples in the specs, but what do you think?)

Additionally we could increase the margin on the right of the button such that it is right aligned with the messages.
![image](https://user-images.githubusercontent.com/52440189/217220096-f1ea76e9-06c4-4b04-92de-dd58f96199d8.png)

Other than that it is great! :)

AndyScherzinger commented:
Hi @nimishavijay, 

see https://github.com/nextcloud/talk-android/pull/2741#issue-1573995363 I changed 

* the right padding and 
* added a subtle drop-shadow for the elevation.

I would keep it circular-rounded - not like a fab - because it is technically not a fab, also checked signal, telegram, whatsapp and Google messages and they all use a circular button like we do for the scroll-to-bottom.

mahibi commented on app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt:
```diff
@@ -674,6 +674,7 @@ class ChatController(args: Bundle) :
         }
 
         binding?.scrollDownButton?.setOnClickListener { binding?.messagesListView?.smoothScrollToPosition(0) }
```
@AndyScherzinger 
smoothScrollToPosition is a bit jerky for me.
i suggest we change smoothScrollToPosition to scrollToPosition which doesn't have the automated scrolling but feels smoother UX wise..?

AndyScherzinger replied:
Yes, just did via https://github.com/nextcloud/talk-android/pull/2741/commits/45f6e3b47426d6dd1fa6bd869adace8c3f839525 - also check whatsapp and they also do it like you suggested @mahibi 👍
